presto-main/src/main/java/com/facebook/presto/execution/TaskExecutor.java

```diff
@@ -855,6 +888,20 @@ public long getRunningTasksLevel4()
     }
 
     @Managed
+    public long getLongestSplitTimeMillis()
+    {
+        Iterator<SplitProfile> iterator = splitProfiler.iterator();
```

Comment:
Controlling execution with exception is known as a bad practice. IMO approach with iterator was much better.

I have heard that exception handling was fixed in java, and it is no longer causing big performance degradation, but still I would use exceptions only to exceptional situations. 
In case you except that this collection should always have at least one element, then checkState should be added.